### PR TITLE
Fix sample code and add READMEs

### PR DESCRIPTION
**Pull request by dsyer**
The typed-component sample didn't work at all. And both that 
and the containerized-workload sample (as mentioned in top
level README) lacked any documentation to indicate how to
use them and verify that they were working correctly.

### CODE REVIEW

**Comment by dsyer** [comment]
I'm not sure the `typed-component` is working as designed even with this change. It seems to make no difference if I delete the `workload.spec.type`. So I guess it's actually not doing anythin?

File: examples/typed-component/sample_component.yaml
Comment (resouer) [inline examples/typed-component/sample_component.yaml:8]: `type` is mutually exclusive with `apiVersion` and `kind`, so please leave this yaml as is. Typed component is a experimental design in oam-runtime and it's not compliance with OAM spec v0.2.1.

Also, we decide not to accept this format in OAM spec per recent community meeting, so `type` will be deprecated soon.

File: examples/typed-component/README.md
Comment (resouer) [inline examples/typed-component/README.md:41]: Would you mind to change the image to sth else like `nginx`? Wordpress doesn't make sense to me unless a database component is included.

**Review by resouer — COMMENTED** [review]
Thank you Deve, some comments left.

File: examples/typed-component/sample_component.yaml
Comment (dsyer) [reply examples/typed-component/sample_component.yaml:8]: If type is deprecated we should remove it though? That’s the opposite. So don’t know what to do.

File: examples/typed-component/README.md
Comment (dsyer) [reply examples/typed-component/README.md:41]: It’s not my sample. I just used what was there.

File: examples/typed-component/sample_component.yaml
Comment (resouer) [reply examples/typed-component/sample_component.yaml:8]: Yes, exactly, it should be removed. That's why I suggest leave it as is unless you want to take this burden (I don't think so, and it's safer to do clean up in separate PR).

File: examples/typed-component/README.md
Comment (resouer) [reply examples/typed-component/README.md:41]: LGTM then, you've already done your part!

**Comment by resouer** [comment]
LGTM when `typed-component` section is restored. A follow up PR will be send to clean up the samples. /cc @dsyer

**Review by resouer — APPROVED** [review]
LGTM after typed-component section is restored. Maintainers will send follow up PR to clean up the samples. /cc @dsyer

File: examples/typed-component/sample_component.yaml
Comment (wonderflow) [inline examples/typed-component/sample_component.yaml:7]: the typed component can work if we enable webhook of oam-k8s-runtime

**Comment by dsyer** [comment]
Restored component definition and put a note about it in the README. That should resolve all the comments above.

**Review by wonderflow — APPROVED** [review]
Thanks! @dsyer

**Pull request merged**
